### PR TITLE
Disallow unexpect console output during tests and fix all errors

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Similar to #2041 (which this PR builds upon), this PR prevents any unexpected console errors from appearing in tests via the `jest-fail-on-console` utility. This prevents people observing test runs from being confused as to why errors are appearing in the test logs. It also forces us to actually fix these errors instead of glossing over them or ignoring them.

## Description

Introduce `jest-fail-on-console` to prevent unexpected console logging in tests. Please note that with this utility you can still enabled _expected_ console logging of any level and the utility gives clear instructions for how to do so when tests fail due to unexpected console logs.

This PR also then fixes all the instances where tests were logging errors. Mostly this was through fixing a variety of fixtures and test configurations.

## Testing Instructions

Run the unit tests and confirm that they pass (you can rely on CI for this). Also check the CI output to confirm that there are no logged warnings or errors of any kind in the tests (indeed, they should fail if there are).

Check out the branch and add a console error to a test locally to test that the `jest-fail-on-console` utility is configured correctly.

Finally, because I had to make changes to the focus trap, test some dialogues in the running application to make sure they still work as expected. Specifically test that when opening them via keyboard navigation that the correct element is focused when the dialogue is opened.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/test/unit/setup-after-env.js [hunk 1]
```diff
@@ -1,5 +1,8 @@
 import Vue from "vue"
 import "@testing-library/jest-dom"
+import failOnConsole from "jest-fail-on-console"
+
+failOnConsole()
```
Comment (dhruvkb) [inline frontend/test/unit/setup-after-env.js:5]: Will this also catch output from our custom console utils? They indirectly refer to the inbuilt `console` functions but I'm asking just to be sure.

**Review by dhruvkb — APPROVED** [review]
Thanks for fixing a bunch of these errors. LGTM, except one doubt.

File: frontend/test/unit/setup-after-env.js (hunk 1 above)
Comment (sarayourfriend) [reply frontend/test/unit/setup-after-env.js:5]: Yup! The library is implemented by modifying the global console object so unless we were doing something like writing directly to `stdout`, we wouldn't be able to easily _accidentally_ bypass this utility.

https://github.com/ValentinH/jest-fail-on-console/blob/main/index.js#L102-L115

**Comment by sarayourfriend** [comment]
Closing as @obulat has incorporated these changes into #2044

**Pull request closed without merging**